components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/listener/TenantCreationEventListener.java

```diff
@@ -101,6 +102,12 @@ public void onPreDelete(int tenantId) throws StratosException {
         } catch (IdentityOAuthAdminException e) {
             throw new StratosException("Error in deleting all OAuth application data of the tenant: " + tenantId, e);
         }
+
```

Comment:
This should be done in framework level? Didn't we delete the applications in a framework listener?